src/validators/var_kwargs.rs

```diff
@@ -0,0 +1,77 @@
+use std::str::FromStr;
+
+use pyo3::intern;
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyString};
+
+use crate::input::Input;
+use crate::tools::SchemaDict;
+
+use super::validation_state::ValidationState;
+use super::{build_validator, BuildValidator, CombinedValidator, DefinitionsBuilder, Validator};
+
+enum VarKwargsMode {
+    Single,
+    TypedDict,
+}
+
+impl FromStr for VarKwargsMode {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "single" => Ok(VarKwargsMode::Single),
+            "typed_dict" => Ok(VarKwargsMode::TypedDict),
+            _ => Err(()),
+        }
+    }
+}
```

Comment:
```suggestion
impl From<&str> for VarKwargsMode {
    fn from_str(s: &str) -> Self {
        match s {
            "single" => Ok(VarKwargsMode::Single),
            "typed_dict" => Ok(VarKwargsMode::TypedDict),
        }
    }
}
```

Not sure what's the best convention to use here?

Reply:
`FromStr` is the correct choice

Reply:
In which case you would chose `From<&str>`? This one doesn't return a `Result`, but I'm wondering in which case this is preferable

Reply:
I would use `From` if the parsing can never fail. In this case it can if it's not one of the two specific cases.